app/javascript/components/CaseNoteForm.jsx

```diff
@@ -0,0 +1,357 @@
+  Switch,
+} from '@material-ui/core/';
+import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles';
+import MUIRichTextEditor from 'mui-rte';
+
+const styles = {
+  dialogActionsStyle: {
+    padding: '30px',
+  },
+  MUIRichTextEditorStyle: {
+    border: '5px solid',
+    padding: '10px',
+  },
+  dialogStyle: {
+    padding: '20px',
+  },
+  dialogContentTextStyle: {
+    color: 'black',
+    marginBottom: '2px',
+  },
+  dialogContentTextFieldStyle: {
+    marginTop: '2px',
+    borderStyle: 'solid 4px grey',
+  },
+  saveDocumentButtonStyle: {
+    borderStyle: 'solid 3px grey',
+  },
+};
+
+const defaultTheme = createMuiTheme();
+
+Object.assign(defaultTheme, {
+  overrides: {
+    MUIRichTextEditor: {
+      root: {
+        border: 'solid 1px #C4C4C4',
+        borderRadius: '4px',
+      },
+      editor: {
+        padding: '20px',
+      },
+    },
+  },
+});
+
+class CaseNoteForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: this.props.title,
+      description: this.props.description,
+      participant_id: this.props.participantId,
+      internal: true,
+      open: false,
+      type: this.props.type,
+      id: this.props.id,
+      tempTitle: this.props.title,
+      tempDescription: this.props.description,
+      tempInternal: this.props.internal,
+    };
+    this.onChange = editorState => this.setState({ editorState });
+    this.handleClose = this.handleClose.bind(this);
+    this.handleOpen = this.handleOpen.bind(this);
+    this.handleInternalChange = this.handleInternalChange.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleOpen() {
+    this.setState({ open: true });
+  }
+
+  handleClose() {
+    this.setState({ open: false });
+  }
+
+  handleChange = name => event => {
+    const { value } = event.target;
+    this.setState({ [name]: value });
+  };
+
+  handleInternalChange = name => event => {
+    this.setState({ [name]: !this.state.internal });
+  };
+
+  handleDescriptionChange = name => state => {
+    // TODO: the line below is the rtf representation. Update to this once rtf on /casenotes
+    const value = JSON.stringify(convertToRaw(state.getCurrentContent()));
+    //const value = state.getCurrentContent().getPlainText();
+    this.setState({ [name]: value });
+  }
+
+  handleSubmit() {
+    if (this.state.type === "create") {
+      const body = {
+        title: this.state.title,
+        description: this.state.description,
+        internal: this.state.internal,
+        participant_id: this.props.participantId,
+      };
+      apiPost('/api/case_notes', { case_note: body })
+        .then(() => window.location.reload())
+        .catch(error => console.error(error));
+    } else if (this.state.type === "edit") {
+      const newTitle = this.state.tempTitle;
```

Comment:
I don't think these temporary variables are necessary (on cancel, you can just clear the state to it's original content)